src/text-buffer.coffee

```diff
@@ -1199,19 +1200,33 @@ class TextBuffer
   ###
 
   # Public: Save the buffer.
-  save: ->
-    @saveAs(@getPath())
+  save: (options) ->
+    @saveAs(@getPath(), options)
 
   # Public: Save the buffer at a specific path.
   #
   # * `filePath` The path to save at.
-  saveAs: (filePath) ->
+  saveAs: (filePath, options) ->
     unless filePath then throw new Error("Can't save buffer with no file path")
 
     @emitter.emit 'will-save', {path: filePath}
     @emit 'will-be-saved', this if Grim.includeDeprecatedAPIs
     @setPath(filePath)
-    @file.writeSync(@getText())
+
+    if options?.backup and @file.existsSync()
+      backupFilePath = filePath + '~'
+      backupFilePath += '~' while fs.existsSync(backupFilePath)
+      fs.writeFileSync(backupFilePath, @file.readSync())
+
+    try
+      @file.writeSync(@getText())
+      fs.removeSync(backupFilePath) if backupFilePath?
+    catch error
+      if backupFilePath?
+        fs.writeFileSync(filePath, fs.readFileSync(backupFilePath))
```

Comment:
Right, that makes sense.  I just assumed you were doing something different and read it my way.  :)

I figured you were writing the *new* contents to the backup file...  That way, if the main write fails, you still have the new contents in the backup file.  And, if the backup write fails, you still have the old contents in the main file.  When users hit save, they don't care about the old contents anymore, right?

Either way sounds fine to me.